prov/shm/src/smr_init.c

```diff
@@ -72,6 +72,48 @@ static void smr_resolve_addr(const char *node, const char *service,
 	(*addr)[*addrlen - 1]  = '\0';
 }
 
+static uint64_t smr_calculate_shm_usage(size_t tx_count, size_t rx_count)
+{
+	size_t total_size;
+	size_t tx_size, rx_size;
+
+	tx_size = roundup_power_of_two(tx_count);
+	rx_size = roundup_power_of_two(rx_count);
+
+	total_size = sizeof(struct smr_region) +
+		     sizeof(struct smr_cmd_queue) + sizeof(struct smr_cmd) * rx_size +
+		     sizeof(struct smr_resp_queue) + sizeof(struct smr_resp) * tx_size +
+		     sizeof(struct smr_inject_pool) + sizeof(struct smr_inject_pool_entry) * rx_size +
+		     sizeof(struct smr_sar_pool) + sizeof(struct smr_sar_pool_entry) * SMR_MAX_PEERS +
+		     sizeof(struct smr_peer_data) * SMR_MAX_PEERS +
+		     NAME_MAX;
+	total_size = roundup_power_of_two(total_size);
+	return total_size * SMR_MAX_PEERS;
+}
+
+static int smr_shm_space_check(size_t tx_count, size_t rx_count)
+{
+	struct statvfs stat;
+	char shm_fs[] = "/dev/shm";
+	uint64_t size, shm_usage;
+	int err;
+
+	shm_usage = smr_calculate_shm_usage(tx_count, rx_count);
+	err = statvfs(shm_fs, &stat);
+	if (err) {
+		FI_WARN(&smr_prov, FI_LOG_CORE,
+			"Get filesystem %s statistics failed %d (%s)\n",
+			shm_fs, err, fi_strerror(-err));
+		return err;
```

Comment:
return -errno for consistency.  err is just -1.  Need to update fi_strerror() to be strerror(errno)

Reply:
Thank you. Will fix it.